api-2.2/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl_2_2.java

```diff
@@ -42,6 +52,47 @@ public Condition getConditionByUuid(String uuid) {
 		        .uniqueResult();
 	}
 	
+	private ConditionClinicalStatus convertStatus(String status) {
+		if ("active".equalsIgnoreCase(status)) {
+			return ConditionClinicalStatus.ACTIVE;
+		}
+		if ("inactive".equalsIgnoreCase(status)) {
+			return ConditionClinicalStatus.INACTIVE;
+		}
+		// TODO during review: What value of the spec. should be mapped to HISTORY_OF?
+		// http://www.hl7.org/fhir/valueset-condition-clinical.html
+		return ConditionClinicalStatus.HISTORY_OF;
+	}
+	
+	@Override
+	public Collection<Condition> searchForConditions(ReferenceAndListParam patientParam, ReferenceAndListParam subjectParam,
+	        TokenOrListParam code, TokenOrListParam clinicalStatus, DateParam onsetDate, QuantityParam onsetAge,
+	        DateParam recordedData, SortSpec sort) {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Condition.class);
+		
+		handlePatientReference(criteria, patientParam);
+		// TODO during review: Do we require a ":Patient" modifier or a "Patient/" type for the "subject" search param?
+		// And do we want to support anything other than "Patient" under "subject" based searches (e.g., "Group")?
+		handlePatientReference(criteria, subjectParam);
+		handleDate("onsetDate", onsetDate).ifPresent(criteria::add);
+		// TODO: Handle onsetAge as well.
+		handleDate("dateCreated", recordedData).ifPresent(criteria::add);
+		handleOrListParam(clinicalStatus,
+		    tokenParam -> Optional.of(eq("clinicalStatus", convertStatus(tokenParam.getValue())))).ifPresent(criteria::add);
+		if (code != null) {
```

Comment:
Use `handleResourceCode()` for this. We should give that function a more descriptive name... and some docs!

Reply:
Thanks! Yeah... `generateSystemQuery()` was written with some different assumptions. Thanks for fixing that.